src/libs/ReportUtils.ts

```diff
@@ -1173,10 +1173,25 @@ function isJoinRequestInAdminRoom(report: OnyxEntry<Report>): boolean {
     return ReportActionsUtils.isActionableJoinRequestPending(report.reportID);
 }
 
+/**
+ * Checks if the user can write in the provided report
+ */
+function canWriteInReport(report: OnyxEntry<Report>): boolean {
+    if (Array.isArray(report?.permissions) && report?.permissions.length > 0) {
+        return report?.permissions?.includes(CONST.REPORT.PERMISSIONS.WRITE);
+    }
+
+    return true;
```

Comment:
If permissions are not an array or the array length is 0, I think this should return false?

Reply:
I think yes, because it is a new property and I am not sure if it is applied to all kinds of reports. If we return `false` then all legacy chats will be "read-only".
cc @mountiny

Reply:
In Web, we [do not return the permissions](https://github.com/Expensify/Web-Expensify/blob/7780b3ed0e08c5925ef418b5f56b856e9d4a1b1b/lib/ReportUtils.php#L3517-L3564) and we dont return it from the getChats method hence in OpenApp (which uses that method) we also do not include `permissions`. Its only added in the [structureReportForOnyx](https://github.com/Expensify/Auth/blob/fc7ec8622ac06354b4aa1b691428ee5bf5e4087b/auth/lib/Report.cpp#L8559)

Reply:
We however return this in the OpenReport using the [getStructuredOnyxReportAndLastMessageData](https://github.com/Expensify/Auth/blob/fc7ec8622ac06354b4aa1b691428ee5bf5e4087b/auth/command/OpenReport.cpp#L536)

So actually the only one I found that does not have the permissions is the selfDM 🤔 although I see the report should be shared with read, write permissions cc @thienlnam

Reply:
Ah this is because the user owns the report and is so is not in the sharedReports table. Maybe we can an owner check and then populate those permissions in the response